options/rtld/generic/linker.cpp

```diff
@@ -896,6 +902,17 @@ void ObjectRepository::_parseDynamic(SharedObject *object) {
 		object->soName = reinterpret_cast<const char *>(object->baseAddress
 				+ object->stringTableOffset + *soname_offset);
 	}
+	if(object->haveTextRel) {
+		for(auto &phdr_ptr : object->exec_phdrs) {
+			elf_phdr* phdr = (elf_phdr *)phdr_ptr;
+			void* addr = (void *)(phdr->p_vaddr + (uintptr_t)object->baseAddress);
+			int prot = PROT_WRITE | PROT_READ | PROT_EXEC;
```

Comment:
We should never map as RWX. Instead, map as RW first and then change that to RX.